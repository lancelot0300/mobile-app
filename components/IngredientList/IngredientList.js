import {
  Text,
  View,
  FlatList,
  TouchableOpacity,
} from "react-native";
import React from "react";


export const IngredientList = ({ activeIngredient, setActive }) => {

  return (
    <View style={{ height: 60 }}>
      { activeIngredient ?
        Object.keys(activeIngredient).length > 0 && 
        <FlatList
          data={Object.keys(activeIngredient)}
          renderItem={({ item }) => (
            <TouchableOpacity
              style={{
                backgroundColor: "#121212",
                padding: 10,
                margin: 10,
                borderRadius: 18,
                minWidth: 50,
                borderWidth: 1,
                height: 40,
                borderColor: activeIngredient[item]
                  ? "#f954ba"
                  : "rgba(255,255,255,0.25)",
              }}
              onPress={() => {
                setActive({
                  ...activeIngredient,
                  [item]: !activeIngredient[item],
                });
              }}
            >
              <Text
                style={{
                  color: activeIngredient[item] ? "#f954ba" : "#fff",
                  textAlign: "center",
                }}
              >
                {item}
              </Text>
            </TouchableOpacity>
          )}
          keyExtractor={(item) => item}
          horizontal={true}
        />
        : <Text>Loading</Text>
        }
    </View>
  );
};
