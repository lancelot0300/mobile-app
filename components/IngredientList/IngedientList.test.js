import React from 'react';
import renderer from 'react-test-renderer';

import {IngredientList} from './IngredientList';

describe('<IngredientList />', () => {

    it('renders correctly', () => {
        const tree = renderer.create(<IngredientList activeIngredient={{
            "ingedient1": "true",
            "ingedient2": "false",
            "ingedient3": "true",
        }} />).toJSON();
        expect(tree).toMatchSnapshot();
    }
    );

    it('showing loading text', () => {
        const tree = renderer.create(<IngredientList />).toJSON();
        expect(tree.children[0].children[0]).toBe('Loading');
    }
    );

    it('showing first ingredients ', () => {
        const tree = renderer.create(<IngredientList activeIngredient={{
            "ingedient1": "true",
            "ingedient2": "false",
            "ingedient3": "true",
        }} />).toJSON();
        expect(tree.children[0].children[0].children[0].children[0].children[0].children[0]).toBe('ingedient1');
    });
});