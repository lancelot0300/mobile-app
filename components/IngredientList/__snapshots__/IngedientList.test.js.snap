// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IngredientList /> renders correctly 1`] = `
<View
  style={
    {
      "height": 60,
    }
  }
>
  <RCTScrollView
    data={
      [
        "ingedient1",
        "ingedient2",
        "ingedient3",
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#121212",
              "borderColor": "#f954ba",
              "borderRadius": 18,
              "borderWidth": 1,
              "height": 40,
              "margin": 10,
              "minWidth": 50,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#f954ba",
                "textAlign": "center",
              }
            }
          >
            ingedient1
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#121212",
              "borderColor": "#f954ba",
              "borderRadius": 18,
              "borderWidth": 1,
              "height": 40,
              "margin": 10,
              "minWidth": 50,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#f954ba",
                "textAlign": "center",
              }
            }
          >
            ingedient2
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#121212",
              "borderColor": "#f954ba",
              "borderRadius": 18,
              "borderWidth": 1,
              "height": 40,
              "margin": 10,
              "minWidth": 50,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#f954ba",
                "textAlign": "center",
              }
            }
          >
            ingedient3
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
